src/providers/bttv/BttvEmotes.cpp

```diff
@@ -138,16 +139,41 @@ void BttvEmotes::loadEmotes()
         .execute();
 }
 
-void BttvEmotes::loadChannel(const QString &channelId,
+void BttvEmotes::loadChannel(TwitchChannel &channel, const QString &channelId,
```

Comment:
We're passing in a reference to TwitchChannel here, which will be sent to an asynchronous function. We have no way of ensuring that the TwitchChannel is still alive once we get the response from the network call.

The way this is solved can be seen in the TwitchChannel::refreshXXXEmotes call, where we send in a weak pointer to the channel instead of a straight up reference.

Reply:
Fixed.